### PR TITLE
Slackbot role to recipients mapping

### PR DESCRIPTION
**Pull request by Joerger**
### Changes
 - Add support for `role_to_recipients` configuration option.
 - Deprecate `slack.recipients` since its functionality is equivalent to `role_to_recipients["*"]`.
   - `slack.recipients` is still supported, but it can't be used at the same time as `role_to_recipients`, and using it will result in a deprecation warning message. It will be removed in v11.0.0.
 - Update `README.md`
 - Update Teleport Docs - https://github.com/gravitational/teleport/pull/10607
   - Removed `INSTALL.md` which is unmaintained and a copy of the docs.
   - Brought API configuration docs up to date.
   - Replaced docs for `slack.recipients` with `role_to_recipients`.

### Usage

```toml
[teleport] 
addr = "proxy.example.com:443"
identity = "/var/lib/identity"

[slack]
token = "token"

[role_to_recipients]
"dev" = "devs-slack-channel"
"*" = ["admin@email.com", "admin-slack-channel"]
```

Note that `role_to_recipients` uses a custom marshaler to allow both `string` and `[]string` values.

Closes https://github.com/gravitational/teleport-plugins/issues/411
Closes https://github.com/gravitational/teleport-plugins/issues/279

### CODE REVIEW

File: access/slack/config.go [hunk 1]
```diff
@@ -1,56 +1,62 @@
...
 	"github.com/gravitational/teleport-plugins/lib"
 	"github.com/gravitational/teleport-plugins/lib/logger"
+	"github.com/gravitational/teleport/api/types"
 	"github.com/gravitational/trace"
 	"github.com/pelletier/go-toml"
 )
 
 // Config stores the full configuration for the teleport-slack plugin to run.
 type Config struct {
-	Teleport lib.TeleportConfig `toml:"teleport"`
```
Comment (Joerger) [inline access/slack/config.go:14]: providing the `toml:"..."` explicitly is only needed for non-lower-camel-case options. e.g. `role_to_recipients`.

File: access/slack/config.go [hunk 2]
```diff
@@ -1,56 +1,62 @@
...
+	Log        logger.Config
 }
 
 // SlackConfig holds Slack-specific configuration options.
 type SlackConfig struct {
-	Token      string   `toml:"token"`
-	Recipients []string `toml:"recipients"`
+	Token string
+	// DELETE IN 10.0.0 (Joerger) - use "role_to_recipients["*"]" instead
+	Recipients []string
```
Comment (tcsc) [inline access/slack/config.go:26]: Should we show  a deprecation message when this is configured?

**Review by tcsc — APPROVED** [review]

**Comment by Joerger** [comment]
@xinding33 can you check the product decision section when you get a chance?

**Comment by r0mant** [comment]
@Joerger FYI we discussed this on this week's product meeting and agreed to keeping the old `slack.recipients` for backwards compatibility but deprecating/hiding it from any user-facing material in favor of the new mapping syntax.

**Pull request merged**